ui/src/onboarding/containers/LoginPageContents.tsx

```diff
@@ -0,0 +1,443 @@
+                    />
+                  </animated.div>
+                ))
+              }
+            </Transition>
+            <Transition
+              native
+              reset
+              unique
+              items={loginTabActive === false}
+              from={{height: 0}}
+              enter={[
+                {
+                  position: 'relative',
+                  overflow: 'hidden',
+                  height: 'auto',
+                },
+              ]}
+              leave={{height: 0}}
+            >
+              {show =>
+                show &&
+                (props => (
+                  <animated.div style={props}>
+                    <SignUpForm
+                      buttonStatus={buttonStatus}
+                      confirmPassword={confirmPassword}
+                      confirmPasswordValidation={this.formFieldTypeFactory(
+                        confirmPasswordError
+                      )}
+                      email={email}
+                      emailValidation={this.formFieldTypeFactory(emailError)}
+                      firstName={firstName}
+                      firstNameValidation={this.formFieldTypeFactory(
+                        firstNameError
+                      )}
+                      lastName={lastName}
+                      lastNameValidation={this.formFieldTypeFactory(
+                        lastNameError
+                      )}
+                      password={password}
+                      passwordValidation={this.formFieldTypeFactory(
+                        passwordError
+                      )}
+                      handleInputChange={this.handleInputChange}
+                    />
+                  </animated.div>
+                ))
+              }
+            </Transition>
+          </Panel.Body>
+        </Panel>
+      </form>
+    )
+  }
+
+  private get validateFieldValues(): {
+    isValid: boolean
+    errors: {[fieldName: string]: string}
+  } {
+    const {
+      activeTab,
+      firstName,
+      lastName,
+      email,
+      password,
+      confirmPassword,
+    } = this.state
+
+    const firstNameError = firstName === '' && 'First name is required'
+    const lastNameError = lastName === '' && 'Last name is required'
+    const emailError = email === '' && 'Email is required'
+    const passwordError = password === '' && 'Password is required'
+    const confirmPasswordError =
+      confirmPassword === password
+        ? confirmPassword === '' && 'Confirm password is required'
+        : "The input passwords don't match"
+
+    const errors: ErrorObject = {
+      emailError,
+      passwordError,
+    }
+    if (activeTab === 'signup') {
+      errors.firstNameError = firstNameError
+      errors.lastNameError = lastNameError
+      errors.confirmPasswordError = confirmPasswordError
+    }
+
+    const isValid = !Object.values(errors).some(error => !!error)
```

Comment:
change `some` to `every` to check for an empty string

Reply:
change `isValid` to actually mean that there are no errors, right now it's indicating that there are errors